### PR TITLE
Add charts to home.html

### PR DESCRIPTION
**Pull request by erikgarciam**
-- Add charts to home page.
Screenshot 2020-07-09 at 11 52 12 AM

### CODE REVIEW

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js [hunk 1]
```diff
@@ -10,4 +10,142 @@
...
+
+    if (keywordCampaignId != 0) {
+        DSACampaignsList.innerHTML = '';
+
+        fetch('/DSA-campaigns?keywordCampaignId=' + keywordCampaignId).then(response => response.json()).then(DSACampaigns => {
+
+            for(var i = firstChart; i < secondChart; i++ ){
+                drawNewBarGraph(DSACampaigns[i], secondChart-i);
+            }
+            if (firstChart == 0)
```
Comment (nevingeorge) [inline DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:49]: Can you use curly braces for your if statements (even though they are one-liners)?

File: DSA-Campaign-Uplift-Estimation/target/DSA-Campaign-Uplift-Estimation-1/Compare/compare.html [hunk 2]
```diff
@@ -1,45 +0,0 @@
-<!DOCTYPE html>
```
Comment (nevingeorge) [inline DSA-Campaign-Uplift-Estimation/target/DSA-Campaign-Uplift-Estimation-1/Compare/compare.html]: Why did you delete all the target files? I think this may mess up my index.yaml file in the targets folder - can you instead just remove the target files from the commit?

Comment (erikgarciam) [reply DSA-Campaign-Uplift-Estimation/target/DSA-Campaign-Uplift-Estimation-1/Compare/compare.html]: I think that's outdated

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js (hunk 1 above)
Comment (erikgarciam) [reply DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:49]: You're right It'd look cleaner to remove the braces

Comment (nevingeorge) [reply DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:49]: Oh I mean add the braces not remove the braces? Google likes having braces even though they are one-liners.

File: DSA-Campaign-Uplift-Estimation/target/DSA-Campaign-Uplift-Estimation-1/Compare/compare.html (hunk 2 above)
Comment (nevingeorge) [reply DSA-Campaign-Uplift-Estimation/target/DSA-Campaign-Uplift-Estimation-1/Compare/compare.html]: Ok, hopefully it doesn't mess anything up - if it does though it should be an easy fix so no worries.

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js [hunk 3]
```diff
@@ -10,4 +10,143 @@
 // distributed under the License is distributed on an "AS IS" BASIS,
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
-// limitations under the License.
\ No newline at end of file
+// limitations under the License.
+
+// Sets the page to always display the first two elements when loading the page.
+var secondChart = 2;
```
Comment (avenkat-google) [inline DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:16]: What if we want to display more than two charts?

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js [hunk 4]
```diff
@@ -10,4 +10,143 @@
...
+    data.addColumn('string', 'DSA Campaign');
+    data.addColumn('number', 'Impressions');
+    data.addColumn('number', 'Clicks');
+    data.addColumn('number', 'Cost (USD)');
+
+    data.addRow([DSACampaign.name, DSACampaign.impressions, DSACampaign.clicks, DSACampaign.cost]);
+
+    var options = {
+        chart: {
+            title: 'Statistics',
```
Comment (avenkat-google) [inline DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:116]: Can we get a dynamic title that describes the graph like "Uplift metrics for ____"

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js [hunk 5]
```diff
@@ -10,4 +10,143 @@
...
+            else{
+                makePagination((DSACampaigns.length / 2) + (DSACampaigns.length % 2), (Math.log(firstChart)/Math.log(2)));
+            }            
+        });
+    }
+}
+
+// Creates pagination html for the user to see other charts by taking the existing number of pages
+// and the active page the user is on.
+function makePagination(numberOfPages,activeNumber) {
```
Comment (avenkat-google) [inline DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:61]: nit: space after comma

Comment (avenkat-google) [inline DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:61]: activePageNumber instead of activeNumber?

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js [hunk 6]
```diff
@@ -10,4 +10,143 @@
...
+    const DSACampaignsList = document.getElementById('DSA-campaigns');
+    DSACampaignsList.innerHTML = '<p>First select a keyword campaign.</p>';
+    var keywordCampaignId = document.getElementById("keyword-campaigns").value;
+
+    if (keywordCampaignId != 0) {
+        DSACampaignsList.innerHTML = '';
+
+        fetch('/DSA-campaigns?keywordCampaignId=' + keywordCampaignId).then(response => response.json()).then(DSACampaigns => {
+
+            for(var i = firstChart; i < secondChart; i++ ){
```
Comment (avenkat-google) [inline DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:46]: why not make this dynamic? It is a little strange to have the loop of two charts in this manner, because it implies that firstChart and secondChart may not be one apart.

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js [hunk 7]
```diff
@@ -10,4 +10,143 @@
...
+                drawNewBarGraph(DSACampaigns[i], secondChart-i);
+            }
+            if (firstChart == 0){
+                makePagination((DSACampaigns.length / 2) + (DSACampaigns.length % 2), (Math.log(firstChart + 1)/Math.log(2)));
```
Comment (avenkat-google) [inline DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:50]: Kind of confused about this line why are you taking the log?

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js [hunk 8]
```diff
@@ -10,4 +10,143 @@
...
+            subtitle: 'Impressions, Clicks, and Cost',
+        },
+        bars: 'horizontal' // Required for Material Bar Charts.
+    };
+
+    var chart = new google.charts.Bar(document.getElementById('bar-chart' + chartNumber));
+    chart.draw(data, google.charts.Bar.convertOptions(options));
+    console.log("Drew bar graph.");
+
+    drawNewTable(DSACampaign, chartNumber);
```
Comment (avenkat-google) [inline DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:126]: Remove from this function and have as a standalone.

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js [hunk 9]
```diff
@@ -10,4 +10,143 @@
...
+    if(decreasePage - 1 > 0){
+        decreasePage--;  
+        secondChart -= 2;
+        firstChart -= 2;  
+    }
+    activePage.innerText = decreasePage;
+    getDSACampaigns();
+}
+
+function drawNewBarGraph(DSACampaign, chartNumber) {
```
Comment (avenkat-google) [inline DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:104]: change to drawBarGraphFromDSACampaign or drawDSACampaignBarGraph?

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js [hunk 10]
```diff
@@ -10,4 +10,143 @@
...
+    };
+
+    var chart = new google.charts.Bar(document.getElementById('bar-chart' + chartNumber));
+    chart.draw(data, google.charts.Bar.convertOptions(options));
+    console.log("Drew bar graph.");
+
+    drawNewTable(DSACampaign, chartNumber);
+}
+
+function drawNewTable(DSACampaign, chartNumber) {
```
Comment (avenkat-google) [inline DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:129]: rename this as well.

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js [hunk 11]
```diff
@@ -10,4 +10,143 @@
...
+function previousPage() {
+    const activePage = document.getElementById("activeNumber");
+    var decreasePage = parseInt(activePage.innerText);
+
+    // The if statement checks if the user clicks previousPage but there is
+    // no more charts to look at using 0 because there are no negative pages.
+    if(decreasePage - 1 > 0){
+        decreasePage--;  
+        secondChart -= 2;
+        firstChart -= 2;
```
Comment (avenkat-google) [inline DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:98]: This logic seems a little more complicated than it needs to be. You could just use a page number variable and a constant that represents the number of campaigns worth of charts to render in the page.

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js (hunk 3 above)
Comment (erikgarciam) [reply DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:16]: We agreed that displaying two charts at the same time was better than having the user scroll through the page. Instead the user just clicks next to view the next 2.

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js [hunk 12]
```diff
@@ -10,4 +10,143 @@
...
+        keywordCampaigns.forEach(keywordCampaign => {
+            keywordCampaignList.innerHTML += '<option value=' + keywordCampaign.keywordCampaignId + '>' + keywordCampaign.name  + '</option>';
+        });
+    });
+}
+
+// Checks if user has selected a keyword campaign. If the user has selected a keyword
+// then the for loop draws 2 graphs and creates pagination to explore other charts.
+function getDSACampaigns() {
+    const DSACampaignsList = document.getElementById('DSA-campaigns');
```
Comment (neelbagora) [inline DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:37]: nit: dsaCampaignsList

Comment (erikgarciam) [reply DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:37]: Done.

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js (hunk 10 above)
Comment (erikgarciam) [reply DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:129]: Done.

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js (hunk 9 above)
Comment (erikgarciam) [reply DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:104]: Done.

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js (hunk 8 above)
Comment (erikgarciam) [reply DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:126]: I would leave it here since the graph and table should be made and displayed at the same time.

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js (hunk 7 above)
Comment (erikgarciam) [reply DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:50]: You're right taking the log would cause problems the more it increases. I changed the formula to utilize division and rounding down,

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js (hunk 5 above)
Comment (erikgarciam) [reply DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:61]: Done.

Comment (erikgarciam) [reply DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:61]: Done.

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js [hunk 13]
```diff
@@ -10,4 +10,143 @@
 // distributed under the License is distributed on an "AS IS" BASIS,
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
-// limitations under the License.
\ No newline at end of file
+// limitations under the License.
+
+// Sets the page to always display the first two elements when loading the page.
+var secondChart = 2;
+var firstChart = 0;
```
[the remark this answers is not in the document]
Comment (erikgarciam) [reply DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:17]: Done

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js (hunk 4 above)
Comment (erikgarciam) [reply DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:116]: Done.

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js (hunk 6 above)
Comment (erikgarciam) [reply DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:46]: Fixed.

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js (hunk 11 above)
Comment (erikgarciam) [reply DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:98]: I changed it to fit your description.

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js (hunk 7 above)
Comment (erikgarciam) [reply DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js]: Actually your comment lower regarding, using a page number variable and a constant that represents the number of campaigns made it to where this line of code was changed to just the page number variable.

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js (hunk 8 above)
Comment (avenkat-google) [reply DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js]: In that case can we have a overarching function that makes it clear that we're generating a bar graph and the table. That way it's clear what is happening from the function name, as generating the table as a side-effect of "drawNewBarGraph" is not clear.

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js [hunk 14]
```diff
@@ -10,4 +10,168 @@
...
+    fetch('/keyword-campaigns').then(response => response.json()).then(keywordCampaigns => {
+        keywordCampaigns.forEach(keywordCampaign => {
+            keywordCampaignList.innerHTML += '<option value=' + keywordCampaign.keywordCampaignId + '>' + keywordCampaign.name  + '</option>';
+        });
+    });
+}
+
+// Checks if user has selected a keyword campaign. If the user has selected a keyword
+// then the for loop draws 2 graphs and creates pagination to explore other charts.
+function getDSACampaigns() {
```
Comment (avenkat-google) [inline DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:35]: rename to drawDsaCampaignCharts?

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js [hunk 15]
```diff
@@ -10,4 +10,168 @@
...
+            // chartsToShow checks what the active page and give a starting
+            // point to display the next two charts.
+            var chartsToShow = currentPage * 2;
+
+            // chartCounter keeps track of which number each chart displayed is
+            // (useful for drawing the graph) and is used as a flag to check if 
+            // two charts have been processed.
+            var chartCounter = 1;
+
+            while (chartCounter <= 2) {
```
Comment (avenkat-google) [inline DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:56]: why not just use a for-loop?

Comment (avenkat-google) [inline DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:56]: Make the 2 be a constant named numChartsPerPage, otherwise it seems to be a magic number: https://en.wikipedia.org/wiki/Magic_number_(programming)

Comment (erikgarciam) [reply DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:56]: I felt the while loop was more efficient when it came to setting up the variables for creating the charts.

Comment (erikgarciam) [reply DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:56]: Thanks for the link!. Done.

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js (hunk 14 above)
Comment (erikgarciam) [reply DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js:35]: Done.
